drivers/sensor/icm42605/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+zephyr_library_sources_ifdef(CONFIG_ICM42605 icm42605.c)
+zephyr_library_sources_ifdef(CONFIG_ICM42605 icm42605_setup.c)
+zephyr_library_sources_ifdef(CONFIG_SPI icm42605_spi.c)
```

Comment:
This isn't quite correct since you need icm42605_spi.c to build only if CONFIG_SPI=y && CONFIG_ICM42605=y.  

Might be better to do:

```
if (CONFIG_ICM42605)
  zephyr_library_sources(icm42605.c icm42605_setup.c)
  zephyr_library_sources_ifdef(CONFIG_SPI icm42605_spi.c)
  zephyr_library_sources_ifdef(CONFIG_ICM42605_TRIGGER icm42605_trigger.c)
endif()
```

Reply:
@galak  Thank you and I fixed like below.
As you told, spi.c and trigger.c should build only if CONFIG_ICM42605=y and SPI/TRIGGER=y

if (CONFIG_ICM42605)
  zephyr_library_sources(icm42605.c)
  zephyr_library_sources(icm42605_setup.c)
  zephyr_library_sources_ifdef(CONFIG_SPI icm42605_spi.c)
  zephyr_library_sources_ifdef(CONFIG_ICM42605_TRIGGER icm42605_trigger.c)
endif()